### PR TITLE
[CWS] Introduce security-profile list and save commands

### PR DESCRIPTION
**Pull request by Gui774ume**
### Motivation

This PR introduces 2 new security-agent commands for the security-profile sub command:
- `security-agent runtime security-profile list` lists the security profile currently held in memory by the security profile manager. adding the `--include-cache` flag will also list the security profiles in cache. Example output:

```
security profiles:
  - name: activity-dump-AlYdzgREwA
    workload_selector:
      image_name: ubuntu
      image_tag: latest
    version: local_profile
    status: anomaly_detection,auto_suppression
    kernel_space:
      loaded: true
      loaded_at: 2023-05-11 15:53:23.974169644 +0200 CEST
      cookie: 4418951264499493539 - 0x3d5344c330416aa3
    anomaly_detection_events: [dns exec]
    instances:
      - container_id: b04d9049ff68007a0d819c42a265bec3ab31798ab8560ec8fdc41e801055da38
        tags: [image_name:ubuntu short_image:ubuntu docker_image:ubuntu]
    activity_tree_stats:
      approximate_size: 3824
      process_nodes_count: 3
      file_nodes_count: 4
      dns_nodes_count: 0
      socket_nodes_count: 0
    tags: [image_name:ubuntu short_image:ubuntu docker_image:ubuntu host:cws service:bonsoir source:runtime-security-agent container_id:c979c9812e0f9a2e82f693f1ef4f98dea4d60e5d270ca9c6804427a3a5c18f5c image_tag:latest]
```
- `security-agent runtime security-profile save --name ubuntu --tag latest` saves the current state of the profile for the provided profile selector to disk. Example output:
```
security profile successfully saved at: /tmp/activity-dump-AlYdzgREwA-31945702.profile
```

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Review by lebauce — APPROVED** [review]

File: cmd/security-agent/subcommands/runtime/security_profile.go [hunk 1]
```diff
@@ -78,3 +91,158 @@ func showSecurityProfile(log log.Component, config config.Component, activityDum
...
+		return fmt.Errorf("unable send request to system-probe: %w", err)
+	}
+	if len(output.Error) > 0 {
+		return fmt.Errorf("security profile list request failed: %s", output.Error)
+	}
+
+	if len(output.Profiles) > 0 {
+		fmt.Println("security profiles:")
+		for _, d := range output.Profiles {
+			printSecurityProfileMessage("  ", d)
```
Comment (spikat) [inline cmd/security-agent/subcommands/runtime/security_profile.go:142]: as this func is only called here with the prefix constant, maybe we can move it inside the func itself?

File: pkg/security/security_profile/profile/manager.go [hunk 2]
```diff
@@ -665,3 +667,72 @@ func (m *SecurityProfileManager) tryAutolearn(profile *SecurityProfile, event *m
 
 	return true, nil
 }
+
+// ListSecurityProfiles returns the list of security profiles
+func (m *SecurityProfileManager) ListSecurityProfiles(params *api.SecurityProfileListParams) (*api.SecurityProfileListMessage, error) {
+	var out api.SecurityProfileListMessage
+
+	for _, p := range m.profiles {
```
Comment (spikat) [inline pkg/security/security_profile/profile/manager.go:675]: there is no need to lock here?

**Review by spikat — APPROVED** [review]
Thanks for the work, only two comments otherwise LGTM :)

File: pkg/security/security_profile/profile/manager.go (hunk 2 above)
Comment (Gui774ume) [reply pkg/security/security_profile/profile/manager.go:675]: Good catch !! thanks :)

File: cmd/security-agent/subcommands/runtime/security_profile.go (hunk 1 above)
Comment (Gui774ume) [reply cmd/security-agent/subcommands/runtime/security_profile.go:142]: done 👍🏻

**Pull request merged**